ironfish/src/memPool/memPool.ts

```diff
@@ -56,8 +56,10 @@ export class MemPool {
   }
 
   *get(): Generator<Transaction, void, unknown> {
-    while (!this.queue.isEmpty()) {
-      const feeAndHash = this.queue.poll()
+    const clone = this.queue.clone()
```

Comment:
There might be potential for issues if the cloned queue desyncs from the main queue, e.g. if a block is connected and transactions get removed from the queue while iterating over the cloned one. Not familiar enough with how the constructed block gets verified to say for sure

Reply:
Yeah, what will happen is that the final mined block will be invalid. The only solution is that we would have to lock the blockchain database in the director when creating a block and getting transactions.